mender-update/daemon/states.cpp

```diff
@@ -366,16 +385,29 @@ void UpdateDownloadState::ParseArtifact(Context &ctx, sm::EventPoster<StateEvent
 	}
 	ctx.deployment.artifact_payload.reset(new artifact::Payload(std::move(exp_payload.value())));
 
-	ctx.deployment.update_module->AsyncDownload(
-		ctx.event_loop, *ctx.deployment.artifact_payload, [&poster](error::Error err) {
-			if (err != error::NoError) {
-				log::Error(err.String());
-				poster.PostEvent(StateEvent::Failure);
-				return;
-			}
+	if (ctx.deployment.download_with_sizes) {
+		ctx.deployment.update_module->AsyncDownloadWithFileSizes(
+			ctx.event_loop, *ctx.deployment.artifact_payload, [&poster](error::Error err) {
+				if (err != error::NoError) {
+					log::Error(err.String());
+					poster.PostEvent(StateEvent::Failure);
+					return;
+				}
 
-			poster.PostEvent(StateEvent::Success);
-		});
+				poster.PostEvent(StateEvent::Success);
+			});
+	} else {
+		ctx.deployment.update_module->AsyncDownload(
+			ctx.event_loop, *ctx.deployment.artifact_payload, [&poster](error::Error err) {
+				if (err != error::NoError) {
+					log::Error(err.String());
+					poster.PostEvent(StateEvent::Failure);
+					return;
+				}
+
+				poster.PostEvent(StateEvent::Success);
```

Comment:
Suggestion: Put this handler in a variable first, then you don't have to specify it twice.

Reply:
Thank you!